core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java

```diff
@@ -157,12 +159,21 @@ private void validateVolumeType(VolumeTemplate value, Platform platform, Validat
         }
     }
 
+    private boolean isIDBrokerInstanceGroup(InstanceGroup instanceGroup) {
+        return GROUP_NAME_ID_BROKER.equalsIgnoreCase(instanceGroup.getGroupName());
+    }
+
     private void validateVolumeCount(VolumeTemplate value, VmType vmType, VolumeParameterType volumeParameterType,
-        ValidationResult.ValidationResultBuilder validationBuilder) {
+        ValidationResult.ValidationResultBuilder validationBuilder, InstanceGroup instanceGroup) {
         if (vmType != null && needToCheckVolume(volumeParameterType, value.getVolumeCount())) {
             VolumeParameterConfig config = vmType.getVolumeParameterbyVolumeParameterType(volumeParameterType);
             if (config != null) {
-                if (value.getVolumeCount() > config.maximumNumber()) {
+                if (isIDBrokerInstanceGroup(instanceGroup)) {
+                    if (value.getVolumeCount() != 0) {
+                        // IDBroker does not use data volume, so its volume count should be zero
+                        validationBuilder.error(String.format("IDBroker volume count: %s is not zero", value.getVolumeCount()));
+                    }
+                } else if (value.getVolumeCount() > config.maximumNumber()) {
                     validationBuilder.error(String.format("Max allowed volume count for '%s': %s", vmType.value(), config.maximumNumber()));
                 } else if (value.getVolumeCount() < config.minimumNumber()) {
```

Comment:
Naive question: Is there possibly any VM type that has a positive minimum size? If so, the change above will cause conflicts if the custom DL is launched with the IDBroker using such an instance type.